sql/core/src/main/scala/org/apache/spark/sql/catalyst/util/V2ExpressionBuilder.scala

```diff
@@ -217,6 +217,9 @@ class V2ExpressionBuilder(e: Expression, isPredicate: Boolean = false) {
       generateExpressionWithName("SUBSTRING", children)
     case Upper(child) => generateExpressionWithName("UPPER", Seq(child))
     case Lower(child) => generateExpressionWithName("LOWER", Seq(child))
+    case BitLength(child) => generateExpressionWithName("BIT_LENGTH", Seq(child))
```

Comment:
does `BIT_LENGTH` in other databases take both string and binary inputs?

Reply:
Yes, you are right, I have researched some databases, both BIT_LENGTH and CHAR_LENGTH functions only support string type, there should be a type restriction here

Reply:
fixed